src/sagemaker/jumpstart/factory/estimator.py

```diff
@@ -0,0 +1,506 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores JumpStart Estimator factory methods."""
+from __future__ import absolute_import
+
+
+from typing import Dict, List, Optional, Union
+from sagemaker import (
+    hyperparameters as hyperparameters_utils,
+    image_uris,
+    instance_types,
+    metric_definitions as metric_definitions_utils,
+    model_uris,
+    script_uris,
+)
+from sagemaker import session
+from sagemaker.async_inference.async_inference_config import AsyncInferenceConfig
+from sagemaker.base_deserializers import BaseDeserializer
+from sagemaker.base_serializers import BaseSerializer
+from sagemaker.debugger.debugger import DebuggerHookConfig, RuleBase, TensorBoardOutputConfig
+from sagemaker.debugger.profiler_config import ProfilerConfig
+from sagemaker.explainer.explainer_config import ExplainerConfig
+from sagemaker.inputs import FileSystemInput, TrainingInput
+from sagemaker.instance_group import InstanceGroup
+from sagemaker.jumpstart.artifacts import _retrieve_kwargs
+from sagemaker.jumpstart.constants import (
+    JUMPSTART_DEFAULT_REGION_NAME,
+    TRAINING_ENTRY_POINT_SCRIPT_NAME,
+)
+from sagemaker.jumpstart.enums import JumpStartScriptScope, KwargUseCase
+from sagemaker.jumpstart.factory import model
+from sagemaker.jumpstart.types import (
+    JumpStartEstimatorDeployKwargs,
```

Comment:
I realize this is annoying, but could we please rename from `*Kwargs` to `*Options` or `*Settings`? I believe this alternate naming is more inline with the fundamental meaning behind the data encapsulated in these classes. It also makes those classes a more natural place to add utility methods.

Reply:
In fact, are we even really getting value out of the factory utilities? They make the `JumpStartEstimator` and `JumpStartModel` class end up just being boilerplate. Couldn't we do all of this logic in the implementation of those classes?

In my opinion, if we wanted to factory then we would need to identify groups of kwargs/options that make sense together (i.e. all the instance type kwargs, all the security setting kwargs, etc.). We can do this as incremental refactoring on the `JumpStartModel` / `JumpStartEstimator` classes

Reply:
I agree that we may want to rename the type, let me think about this